### PR TITLE
Change PDB tests to use pod conditions instead of phase

### PR DESCRIPTION
**Pull request by mortent**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
The PDB test is flaky as described in #86358. I think this is caused by an issue where the e2e tests checks the pod phase to determine if a pod is ready while the pdb controller actually use the pod `Ready` condition for this. This updates the test to use the same check as the PDB controller.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/sig apps
@kubernetes/sig-apps-pr-reviews

### CODE REVIEW

**Review by janetkuo — APPROVED** [review]

**Comment by mortent** [comment]
@soltysh @enisoc Could you take a look at this PR?

File: test/e2e/apps/BUILD [hunk 1]
```diff
@@ -61,6 +61,7 @@ go_library(
         "//staging/src/k8s.io/client-go/tools/cache:go_default_library",
         "//staging/src/k8s.io/client-go/tools/watch:go_default_library",
         "//staging/src/k8s.io/client-go/util/retry:go_default_library",
+        "//staging/src/k8s.io/kubectl/pkg/util/podutils:go_default_library",
```
Comment (spiffxp) [inline test/e2e/apps/BUILD]: I'm not sure picking up the k8s.io/kubectl dep is a good idea, I couldn't find any prior art in test/e2e for importing from it.  Is there something in test/e2e/framework/pod/ we could use?

Comment (mortent) [reply test/e2e/apps/BUILD]: @spiffxp That was a mistake. I have updated the PR to use the right library now.

File: test/e2e/apps/disruption.go [hunk 2]
```diff
@@ -372,9 +373,9 @@ func locateRunningPod(cs kubernetes.Interface, ns string) (pod *v1.Pod, err erro
 			return false, err
 		}
 
-		for i := range podList.Items {
-			if podList.Items[i].Status.Phase == v1.PodRunning {
-				pod = &podList.Items[i]
+		for _, p := range podList.Items {
+			if podutil.IsPodReady(&p) {
+				pod = &p
```
Comment (tnozicka) [inline test/e2e/apps/disruption.go]: is this returning address of a temporary variable?

Comment (mortent) [reply test/e2e/apps/disruption.go]: Yeah, it is taking the address of the iteration variable. It works in this situation due to the early return after the assignment, but we should avoid this. I have updated the PR.

**Pull request merged**